apps/api_web/lib/api_web/rate_limiter/limiter.ex

```diff
@@ -3,14 +3,14 @@ defmodule ApiWeb.RateLimiter.Limiter do
   Behavior for backends to the V3 API rate limiter.
 
   - `start_link(opts)` is called to start the backend by the supervisor.
-  - `rate_limited?(user_id, max_requests)` returns true if the user_id has used too many requests.
+  - `rate_limited?(user_id, max_requests)` returns true if the user_id has used too many requests, along with the number of requests the user_id has used in this time period.
 
   The main option passed to `start_link/1` is `clear_interval` which is a
   number of milliseconds to bucket the requests into.
 
   """
   @callback start_link(Keyword.t()) :: {:ok, pid}
-  @callback rate_limited?(String.t(), non_neg_integer) :: boolean
+  @callback rate_limited?(String.t(), non_neg_integer) :: {boolean, non_neg_integer}
```

Comment:
What about a return_type of `{:remaining, non_neg_integer} | :rate_limited`? It doesn't make sense for there to be a number of requests remaining if you've been limited, so that data is irrelevant.